modules/tinymce/src/plugins/quickbars/test/ts/browser/CommandsTest.ts

```diff
@@ -0,0 +1,36 @@
+import { TestStore } from '@ephox/agar';
+import { describe, it } from '@ephox/bedrock-client';
+import { Arr } from '@ephox/katamari';
+import { TinyHooks } from '@ephox/wrap-mcagar';
+
+import Editor from 'tinymce/core/api/Editor';
+import QuickbarsPlugin from 'tinymce/plugins/quickbars/Plugin';
+
+describe('browser.tinymce.plugins.quickbars.CommandsTest', () => {
+  Arr.each([ true, false ], (withPresetedCommand) => {
+    const store = TestStore();
+    const hook = TinyHooks.bddSetupLight<Editor>({
+      plugins: 'quickbars',
+      inline: true,
+      toolbar: false,
+      menubar: false,
+      setup: (ed: Editor) => {
+        if (withPresetedCommand) {
+          ed.addCommand('QuickbarInsertImage', () => store.add('QII'));
+        }
+      },
+      base_url: '/project/tinymce/js/tinymce'
+    }, [ QuickbarsPlugin ], true);
+
+    it(`TINY-11399: QuickbarInsertImage should be not overwrite by the plugin if it already exists${withPresetedCommand ? ' (withPresetedCommand)' : ''}`, async () => {
```

Comment:
I think if you use the test function pattern here instead then the labels would be less confusing since then you could do. It makes it easier to just define two separate strings. With a loop and a state boolean then you need to use a ternary to me at least that is harder to read.

```
it('Should execute predefined command', () => testQuickInsertImage(true));
it('Should execute overwritten command', () => testQuickInsertImage(false));
```

Reply:
It also adds more nesting if you have context in content and a loop it becomes pretty deep after a while not the case here but in general.

Reply:
the function would need to have the definition of the hook inside it, is there a way to define it and change the setup when you call: `const editor = hook.editor();`?